drivers/adc/ad4858/ad4858.h

```diff
@@ -285,6 +303,10 @@ struct ad4858_init_param {
 	uint16_t chn_or[AD4858_NUM_CHANNELS];
 	/** Channel underrange limit value */
 	uint16_t chn_ur[AD4858_NUM_CHANNELS];
+	/** Channel sleep value */
+	enum ad4858_ch_sleep_value chn_sleep_value[AD4858_NUM_CHANNELS];
```

Comment:
This dev struct would look a lot better if all the channel specific fields would be moved in a different structure:

```
struct ad4858 {
    enum ad4858_chn_softspan chn_softspan;
    uint32_t chn_offset;
    ...
};

struct ad4858_dev {
    struct ad4858 ch[AD4858_NUM_CH];
    ...
};
```

As such, you would separate device and channel state. I'm fine if you don't do this now, but it might help for further extending and maintaining the driver.

Reply:
I don't think we will be adding more members in future.